app/lib/features/pins/providers/pins_provider.dart

```diff
@@ -37,3 +39,22 @@ final pinEditProvider = StateNotifierProvider.family
     .autoDispose<PinEditNotifier, PinEditState, ActerPin>(
   (ref, pin) => PinEditNotifier(pin: pin, ref: ref),
 );
+
+final pinAttachmentManagerProvider =
+    FutureProvider.family.autoDispose<AttachmentsManager, ActerPin>(
+  (ref, acterPin) async {
+    final manager = await acterPin.attachments();
+    final liveManager = ref.watch(attachmentsManagerProvider(manager));
+    return liveManager;
+  },
+);
+
+final pinAttachmentsProvider = FutureProvider.family
+    .autoDispose<List<Attachment>, ActerPin>((ref, acterPin) async {
+  final manager = await acterPin.attachments();
+  final liveManager = ref.watch(attachmentsManagerProvider(manager));
+  return (await liveManager.attachments()).toList();
+});
+
+final selectedPinAttachmentsProvider =
```

Comment:
what is this? This looks like introducing some global state again that is somehow somewhat managed from the outside and that if a widget was being reused could easily point to the wrong thing. Why is this needed? Why isn't the above `pinAttachmentsProvider` enough?

Reply:
This provider is used for selected attachments UI i.e. which aren't yet sent across the wire. This is useful for the UI state.